consumers/jira_c/gojira/jirautils.go

```diff
@@ -0,0 +1,162 @@
+}
+
+// getConfig parses the configuration from config.yaml and returns a config struct
+func getConfig() config {
+	configFile, err := ioutil.ReadFile(os.Getenv(EnvConfigPath))
+	if err != nil {
+		log.Printf("Error while reading config file:   #%v ", err)
+	}
+
+	var newConfig config
+	err = yaml.Unmarshal(configFile, &newConfig)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+	return newConfig
+}
+
+// parseDraconMessage returns a hashmap of the parsed dracon message
+func parseDraconMessage(message string) (map[string]string, error) {
+	jBytes := []byte(message)
+	var draconResult map[string]string
+	err := json.Unmarshal(jBytes, &draconResult)
+	return draconResult, err
+}
+
+// makeCustomField returns the appropriate interface for a jira CustomField given it's type and values
+// :param fieldType: the type of the field in Jira (single-value, multi-value, float)
+// :param values: list of values to be filled in
+// :return the appropriate interface for a CustomField, given the corresponding fieldType and value(s)
+func makeCustomField(fieldType string, values []string) interface{} {
+	switch fieldType {
+	case "single-value":
+		return map[string]string{"value": values[0]}
+	case "multi-value":
+		cf := []map[string]string{}
+		for _, v := range values {
+			cf = append(cf, map[string]string{"value": v})
+		}
+		return cf
+	case "float":
+		if f, err := strconv.ParseFloat(values[0], 64); err == nil {
+			return f
+		} else {
+			log.Fatalf("Error parsing float field-type: %v", err)
+		}
+	default:
+		return nil
+	}
+	return nil
+}
+
+// makeProjectField returns a jira.Project object with the given key value
+func makeProjectField(value string) jira.Project {
+	return jira.Project{
+		Key: value,
+	}
+}
+
+// makeIssueTypeField returns a jira.IssueType object with the given name value
+func makeIssueTypeField(value string) jira.IssueType {
+	return jira.IssueType{
+		Name: value,
+	}
+}
+
+// makeComponentsField returns a []jira.Components object with the given name value(s)
+func makeComponentsField(values []string) []*jira.Component {
+	components := []*jira.Component{}
+	for _, v := range values {
+		components = append(components, &jira.Component{Name: v})
+	}
+	return components
+}
+
+// makeAffectsVersionsField returns a []jira.AffectsVersion object with the given name value(s)
+func makeAffectsVersionsField(values []string) []*jira.AffectsVersion {
+	affectsVersions := []*jira.AffectsVersion{}
+	for _, v := range values {
+		affectsVersions = append(affectsVersions, &jira.AffectsVersion{Name: v})
+	}
+	return affectsVersions
+}
+
+// makeDescription creates the description of an issue's enhanced with extra information from the Dracon Result
+// depending on the tool used, diffrent fields are more suitable as description
+func makeDescription(draconResult map[string]string, extras []string) string {
+	desc := "This issue was automatically generated by the Dracon security pipeline.\n\n"
+	switch draconResult["tool_name"] {
```

Comment:
maybe a future feature would be to have this mapping as YAML configuration that you read in; right now it's tying support to only `gosec`, `spotbugs`, `bandit`. Adding more would require changing code and creating a new release etc.

Reply:
Will work on implementing this in the future